requests/src/main/scala/weco/api/requests/services/SierraRequestsService.scala

```diff
@@ -71,6 +75,14 @@ class SierraRequestsService(
       //      so we look to see if this item can be requested.
       //    - The user's patron record has expired and they can no longer make requests.
       //
+      case Left(SierraErrorCode(132, 2, 500, _, Some(description)))
+          if description.contains("You may not make requests") =>
+        checkIfUserIsSelfRegistered(patron)
+          .map {
+            case Some(rejected) => Left(rejected)
+            case None           => Left(HoldRejected.UserIsSelfRegistered)
+          }
```

Comment:
I think(?) this in combination with `checkIfUserIsSelfRegistered` as implemented below will always return a `HoldRejected.UserIsSelfRegistered`, which isn't what we want - it should be an `UnknownReason` if this error occurs for any patron other than a self-reg patron. I don't believe we'll see users with no pType, but we definitely want to know about it  (in the form of a 500 error) if we do, and anyway the error message above only really makes sense for self-reg users.

That is to say I think the behaviour should be: if we get a 132 error which says 'You may not make requests' _and_ if the patron is self-registered, we should return a `HoldRejected`. Otherwise, we don't know what to say to the user, so we should return an `UnknownReason`.